### PR TITLE
[4.0] [com_tags] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
use prepared statement for SQL

### Testing Instructions
test com_tags

### Expected result
work as before

### Actual result
N/A

### CODE REVIEW

File: components/com_tags/Model/TagsModel.php [hunk 1]
```diff
@@ -103,13 +104,14 @@ protected function getListQuery()
 
 		// Select required fields from the tags.
 		$query->select('a.*, u.name as created_by_user_name, u.email')
```
Comment (Quy) [inline components/com_tags/Model/TagsModel.php:106]: Apply quoteName

File: components/com_tags/Model/TagsModel.php [hunk 2]
```diff
@@ -158,10 +160,13 @@ protected function getListQuery()
...
+			$title = '%' . $this->state->get('list.filter') . '%';
+			$query->where($db->quoteName('a.title') . ' LIKE :title')
+				->bind(':title', $title);
 		}
 
-		$query->where($db->quoteName('a.published') . ' = ' . $published);
+		$query->where($db->quoteName('a.published') . ' = :published')
+			->bind(':published', $published, ParameterType::INTEGER);
 
 		$query->order($db->quoteName($orderby) . ' ' . $orderDirection . ', a.title ASC');
```
Comment (Quy) [inline components/com_tags/Model/TagsModel.php:171]: Apply quoteName

File: components/com_tags/Model/TagsModel.php (hunk 1 above)
Comment (alikon) [reply components/com_tags/Model/TagsModel.php:106]: please submit a new pr for that, 
thanks

File: components/com_tags/Model/TagsModel.php (hunk 2 above)
Comment (alikon) [reply components/com_tags/Model/TagsModel.php:171]: please submit a new pr for that,
thanks

File: administrator/components/com_tags/Model/TagsModel.php [hunk 3]
```diff
@@ -141,53 +142,63 @@ protected function getListQuery()
...
-
-			->select('ug.title AS access_title')
-			->join('LEFT', '#__viewlevels AS ug on ug.id = a.access');
+		$query->select($db->quoteName('ua.name', 'author_name'))
+			->join('LEFT', $db->quoteName('#__users', 'ua'), $db->quoteName('ua.id') . ' = ' . $db->quoteName('a.created_user_id'))
+			->select($db->quoteName('ug.title', 'access_title'))
+			->join('LEFT', $db->quoteName('#__viewlevels', 'ug'), $db->quoteName('ug.id') . ' = ' . $db->quoteName('a.access'));
 
 		// Filter on the level.
 		if ($level = $this->getState('filter.level'))
```
Comment (Quy) [inline administrator/components/com_tags/Model/TagsModel.php]:
```suggestion
		if ($level = (int) $this->getState('filter.level'))
```
Remove line 170.

File: administrator/components/com_tags/Model/TagsModel.php [hunk 4]
```diff
@@ -141,53 +142,63 @@ protected function getListQuery()
...
 		{
-			$query->where('a.level <= ' . (int) $level);
+			$level = (int) $level;
+			$query->where($db->quoteName('a.level') . ' <= :level')
+				->bind(':level', $level, ParameterType::INTEGER);
 		}
 
 		// Filter by access level.
 		if ($access = $this->getState('filter.access'))
```
Comment (Quy) [inline administrator/components/com_tags/Model/TagsModel.php]:
```suggestion
		if ($access = (int) $this->getState('filter.access'))
```
Remove line 178.

File: components/com_tags/Model/TagsModel.php [hunk 5]
```diff
@@ -90,8 +91,8 @@ protected function getListQuery()
 	{
 		$app            = Factory::getApplication();
 		$user           = Factory::getUser();
-		$groups         = implode(',', $user->getAuthorisedViewLevels());
-		$pid            = $this->getState('tag.parent_id');
+		$groups         = $user->getAuthorisedViewLevels();
+		$pid            = (int) $this->getState('tag.parent_id');
 		$orderby        = $this->state->params->get('all_tags_orderby', 'title');
 		$published      = $this->state->params->get('published', 1);
```
Comment (Quy) [inline components/com_tags/Model/TagsModel.php]:
```suggestion
		$published      = (int) $this->state->params->get('published', 1);
```

File: administrator/components/com_tags/Model/TagsModel.php (hunk 3 above)
Comment (alikon) [reply administrator/components/com_tags/Model/TagsModel.php]: done thx

File: administrator/components/com_tags/Model/TagsModel.php (hunk 4 above)
Comment (alikon) [reply administrator/components/com_tags/Model/TagsModel.php]: done thx

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9891b273d31bfb475b4a2c5b6e0322d511f3adc9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26064">issues.joomla.org/tracker/joomla-cms/26064</a>.</sub>

**Comment by joeykrop** [comment]
I have tested this item :white_check_mark: successfully on 0da455def5d4031668c2cb034719f0004bd184ea<br /><br />Tested around with tags in general.

Everything seemed to work as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26064">issues.joomla.org/tracker/joomla-cms/26064</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 0da455def5d4031668c2cb034719f0004bd184ea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26064">issues.joomla.org/tracker/joomla-cms/26064</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26064">issues.joomla.org/tracker/joomla-cms/26064</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**